src/lib/jsonParse/jsonUpdateContextResponse.cpp

```diff
@@ -125,7 +125,7 @@ static std::string attribute(const std::string& path, const std::string& value,
 {
   LM_T(LmtParse, ("Creating an attribute"));
   parseDataP->upcrs.attributeP = new ContextAttribute();
-  parseDataP->upcrs.attributeP->valueType = orion::ValueTypeNone;
+  parseDataP->upcrs.attributeP->valueType = orion::ValueTypeNotGiven;
```

Comment:
Yeah, definitely.
All these assignments can be replaced by one line in the constructor (or, more lines if more constructors need it)

Reply:
Fixed in 707138e0b5ab6fc8c4642afa906290b48fd3f7ae

Not only the ContextAttribute constructor, but also Metatada constructor, with additional fixes in the metadata parsing logic suprisingly unnoticed until now. I'd suggest to review that commit .